src/main/authentication/authenticationHandler.ts

```diff
@@ -8,6 +10,21 @@ export const allAdminRoles = ['SYSTEM_ADMIN', 'INTERNAL_SUPER_ADMIN_CTSC', 'INTE
 export const verifiedRoles = ['VERIFIED'];
 const FRONTEND_URL = process.env.FRONTEND_URL || 'https://pip-frontend.staging.platform.hmcts.net';
 
+export function isAdminSessionExpire(req): boolean {
+  if(checkRoles(req, allAdminRoles)) {
+    if(req.session.sessionExpiry) {
+      const sessionExpiryDateTime = moment.utc(req.session.sessionExpiry);
+      const currentDateTime = moment.utc(new Date(Date.now()));
+      const durationAsSeconds = moment.duration(sessionExpiryDateTime.startOf('seconds').diff(currentDateTime.startOf('seconds'))).asMinutes();
+      if(durationAsSeconds <= 0) {
+        return true;
+      }
+    }
+    req.session.sessionExpiry = new Date(Date.now() + (5000)); //4 hours
```

Comment:
Confused a bit here. How does the 5000 become 4 hours?

Reply:
If you want to add 4 hours to current date time. Might be clearer to do the following I think :)

let dateTime = new Date();
dateTime.setHours(today.getHours() + 4);
req.session.sessionExpiry = dateTime;